src/Raven.Server/Utils/IOExtensions.cs

```diff
@@ -26,6 +26,40 @@ public static void DeleteFile(string file)
             }
         }
 
+        public static bool EnsureReadWritePermissionForDirectory(string directory)
+        {
+            string tmpFileName = null;
+            string testString = "I can write here!";
+            try
+            {
+                if (string.IsNullOrWhiteSpace(directory))
+                {
+                    return false;
+                }
+
+                if (Directory.Exists(directory) == false)
+                {
+                    Directory.CreateDirectory(directory);
+                }
+
+                tmpFileName = Path.Combine(directory, Path.GetRandomFileName());
+                File.WriteAllText(tmpFileName, testString);
+                var read = File.ReadAllText(tmpFileName); //I can read too!
+                return read == testString;
+            }
+            catch
+            {
+                return false;
+            }
+            finally
+            {
+                if (tmpFileName != null)
+                {
+                    File.Delete(tmpFileName);
```

Comment:
Need error handling here, you might fail to write the file, then fail again when trying to delete it.

Reply:
@ayende okay, I was wondering if I should also delete the folder if it was created by the exam code? It is likely that we will write to that folder so it is redundant one hand but it is an unexpected side effect on the other hand.

Reply:
No, don't delete the directory